### ISSUE TITLE
Question about model quantization with multiple inputs and outputs

### ISSUE THREAD

**Issue by onlyinheaven** [openvinotoolkit/nncf#1294]
I tried to use nncf to do Quantization Aware Training，and I followed the tutorial 
> https://docs.openvino.ai/latest/notebooks/302-pytorch-quantization-aware-training-with-output.html#.
     Since my model has multiple inputs and outputs with different size,  I have some questions in the quantization process.
     The config dict I use is as follows:
```
nncf_config_dict = {
    "input_info": 
    [
        {
            "sample_size": [1,1,240,320],
        },
        {
            "sample_size": [1,3,240,320],       
        },
        {
            "sample_size": [1,1,240,320],       
        },
    ],
    
    "log_dir": str(args.logdir),  # The log directory for NNCF-specific logging outputs.
    "compression": {
        "algorithm": "quantization",  # Specify the algorithm here.
        "initializer": {
                    "range": {
                        "num_init_samples": 256
                    }
                }
    },
}
```
   When I deal with the initialization:
```
nncf_config = NNCFConfig.from_dict(nncf_config_dict) 
nncf_config = register_default_init_args(nncf_config, TrainImgLoader)  
```
  I got the error
```
By default it is assumed that the data loader used for initialize produces a tuple/list of (*model_input*, *ground_truth*) and that no special forward arguments have to be set during init. If this is not the case, then instead of your regular data loader you need to pass a specialized version of PTInitializingDataLoader that returns a general (args, kwargs) tuple for your model to be called with at each __next__ call.
```
   Does it means that I should provide a special dataloader for NNCF?Is there any samples I could refer? My inputs in the original dataloader are as : sample{'imgA':imgA, 'imgB':imgB,'imgC': imgC,'GT': GT}
   I also have some questions about the criterion that should be provided to the register_default_init_args(). Since my outputs are two images with different size and I need to calculate L1 loss using the outputs and GT in different scale separately. I don't know how th deal with it.

**Comment by AlexKoff88** [comment]
Hi @onlyinheaven,

Any specific issue you came across? 

If the dataloader you are using is `torch.utils.data.DataLoader` then you just need to pass it to `register_default_init_args(nncf_config, data_loader)`. No extra arguments are required (loss, etc.) in the case of 8-bit quantization. You just wrap the model with NNCF using the `create_compressed_model` call and tune it.
If you need to customize the dataloader for some reason you can refer to this [code](https://github.com/huggingface/optimum-intel/blob/079fb320bb765bbd0d51f690c186d358fd73ece2/optimum/intel/openvino/quantization.py#L53).

**Comment by onlyinheaven** [comment]
> 
Thanks， I will try the customized dataloader
